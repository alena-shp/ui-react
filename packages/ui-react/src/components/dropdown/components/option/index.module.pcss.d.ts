


interface Stylesheet {
    'dropdown-option': string
    'dropdown-option_active': string
}


declare const stylesheet: Stylesheet
export default stylesheet
