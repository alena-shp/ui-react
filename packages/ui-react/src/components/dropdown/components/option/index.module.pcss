.dropdown-option {
    padding: 12px;
    transition: 100ms ease background-color;

    @media (hover: hover) and (pointer: fine) {

        &:hover {
            background-color: var(--color-bg-plate);
            cursor: pointer;
        }
    }

    &_active {
        color: var(--color-text-warning);
    }
}
