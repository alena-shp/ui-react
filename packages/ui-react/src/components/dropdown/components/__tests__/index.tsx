import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';

import { DropdownOption } from '../option';

test('Default use', async () => {
    const { asFragment, getByText } = render(
        <DropdownOption
            onClick={() => jest.fn}
        >
            Content
        </DropdownOption>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Content'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Active option', async () => {
    const { asFragment, getByText } = render(
        <DropdownOption
            isActive={true}
            onClick={() => jest.fn}
        >
            Content
        </DropdownOption>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Content'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

