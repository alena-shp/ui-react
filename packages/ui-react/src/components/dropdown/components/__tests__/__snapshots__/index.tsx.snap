// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active option 1`] = `
<DocumentFragment>
  <div
    class="dropdown-option dropdown-option_active"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Active option 2`] = `
<DocumentFragment>
  <div
    class="dropdown-option dropdown-option_active"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Default use 1`] = `
<DocumentFragment>
  <div
    class="dropdown-option"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Default use 2`] = `
<DocumentFragment>
  <div
    class="dropdown-option"
  >
    Content
  </div>
</DocumentFragment>
`;
